src/components/Provider.js

```diff
@@ -20,39 +22,83 @@ function warnAboutReceivingStore() {
 }
 
 export function createProvider(storeKey = 'store') {
-    const subscriptionKey = `${storeKey}Subscription`
 
     class Provider extends Component {
-        getChildContext() {
-          return { [storeKey]: this[storeKey], [subscriptionKey]: null }
+
+        constructor(props) {
+          super(props)
+
+            const {store} = props;
+
+            this.state = {
+                storeState : store.getState(),
+                store,
+            };
+        }
+
+        componentDidMount() {
+            this.subscribe();
+        }
+
+        componentWillUnmount() {
+          if(this.unsubscribe) {
+            this.unsubscribe()
+            this._isMounted = false;
+          }
         }
 
-        constructor(props, context) {
-          super(props, context)
-          this[storeKey] = props.store;
+        subscribe() {
+          const {store} = this.props;
+
+          this._isMounted = true;
+
+          this.unsubscribe = store.subscribe( () => {
+            const newStoreState = store.getState();
+
+            if(!this._isMounted) {
+              return;
+            }
+
+            this.setState(providerState => {
+              // If the value is the same, skip the unnecessary state update.
+              if(providerState.storeState === newStoreState) {
+                return null;
+              }
+
+              return {storeState : newStoreState};
+            })
+          });
+
+          // Actions might have been dispatched between render and mount - handle those
+          const postMountStoreState = store.getState();
+          if(postMountStoreState !== this.state.storeState) {
+            this.setState({storeState : postMountStoreState});
+          }
         }
 
         render() {
-          return Children.only(this.props.children)
+            return (
+                <ReactReduxContext.Provider value={this.state}>
+                    {Children.only(this.props.children)}
```

Comment:
Children.only is no longer required, no?

Reply:
Hmm.  Y'know, I'm not entirely sure why we even ever had the requirement of a single child to begin with.  Perhaps this is a _very_ old holdover from the earliest prototypes of `connect`.

I don't immediately see a reason why we'd still need to enforce only one child.